libraries/core/src/test/java/com/paritytrading/philadelphia/FIXInitiatorTest.java

```diff
@@ -399,6 +404,21 @@ void receiveTooLongMessage() throws IOException {
                 initiator.receive();
         });
     }
+    
+    @Test
+    @Tag("noTimeInitialization")
+    void firstMessageAfterConnect() throws IOException {
+    	initiator.setCurrentTimeMillis(System.currentTimeMillis());
+    	initiator.keepAlive();
+    	
+    	initiator.sendLogon(false);
+    	
+    	acceptor.receive();
+    	
+    	char firstMessageType = FIXMessage.fromString(acceptorMessages.collect().get(0)).getMsgType().asChar();
+    	
+        assertEquals(FIXMsgTypes.Logon, firstMessageType);        
+    }      
```

Comment:
Could we make this test work so that it would rather just check that if `initiator.keepAlive()` is invoked _before_ `initiator.setCurrentTimeMillis()` is invoked, `initiatorStatus.collect()` would still return an empty list? See e.g. the `heartbeatTimeout` test for reference.

As I understand, the above would have failed before the fix and will now ensure that the issue cannot be accidentally reintroduced.

Reply:
Thank you for your input @jvirtanen.

### Public method:
I thought about the new public method as well, because I don't like to make code worse, just to have a running testcase.

Having the `currentTimestamp` set during the construction of the `FixConnection` leads to a different response in the field 52 than "19700101-00:00:00.000".
Many testcases rely on this "19700101-00:00:00.000", this is why I introduced the new public method, to be able to reset `currentTimestamp` to 0 for the tests.
An exception is the new test `firstMessageAfterConnect()` , which requires a proper time, otherwise, we would have no change compared to the old constructor.

--> The public Method is only removable, if we modify all tests with field 52 to use the current time instead of "19700101-00:00:00.000". This could make the new testcase  `firstMessageAfterConnect()` obsolete and also the simplification below.



### About your suggestion to simplify the testcase, I like your idea to have this:
```
    	initiator.keepAlive();
    	initiator.setCurrentTimeMillis(System.currentTimeMillis());
    	
    	assertEquals(asList(), initiatorStatus.collect());
```

But, sending the test/heartbeat messages in `keepAlive()` will not add messages to initiatorStatus. Only the event "heartbeatTimeout" does, which is called 30 seconds after the first timeout.
So `initiatorStatus` will be empty, equal if the time-variables in constructor are initialized by 0 or by the current time.

You are correct, the testcase is to prevent an accidental reintroduction of the issue.

--> We could modify the `send()` method in `TestConnection` to have access to all sent messages, or keep the test as in the pull request, which I agree isn't a beauty.

Reply:
Good catch regarding the test case. We should check that `initiatorMessages` rather than `initiatorStatus` is empty here. See e.g. the [`heartbeat`](https://github.com/paritytrading/philadelphia/blob/main/libraries/core/src/test/java/com/paritytrading/philadelphia/FIXInitiatorTest.java#L77) test case for an example.